### PR TITLE
Remove unused Moq package from E2E

### PR DESCRIPTION
**Pull request by martinrrm**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2430

Regression? Last working version:

## Description

Removes unused Moq reference in E2E test

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by jeffkl — APPROVED** [review]

**Review by donnie-msft — APPROVED** [review]
Nice find!

File: test/EndToEnd/tests/TabExpansionTest.ps1 [hunk 1]
```diff
@@ -282,7 +282,6 @@ function Test-TabExpansionForVersionForUninstallPackage {
     # Arrange
     $p = New-WebApplication
     $p | Install-Package elmah -Version 1.1
```
Comment (aortiz-msft) [inline test/EndToEnd/tests/TabExpansionTest.ps1:284]: Could we use this opportunity to move all our tests to .NET or Azure SDK packages?

**Review by aortiz-msft — CHANGES_REQUESTED** [review]

File: test/EndToEnd/tests/TabExpansionTest.ps1 (hunk 1 above)
Comment (martinrrm) [reply test/EndToEnd/tests/TabExpansionTest.ps1:284]: Just looked at "elmah" package and its being used 156 times in all kind of files, even in [commmand line outputs ](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.Designer.cs#L715)😨 

Maybe that task is big and we will need to create another task for that, I suggest fixing this asap first

Comment (aortiz-msft) [reply test/EndToEnd/tests/TabExpansionTest.ps1:284]: Could you please create an issue to track that?

**Review by aortiz-msft — APPROVED** [review]

File: test/EndToEnd/tests/TabExpansionTest.ps1 (hunk 1 above)
Comment (martinrrm) [reply test/EndToEnd/tests/TabExpansionTest.ps1:284]: @aortiz-msft Created the issue in https://github.com/NuGet/Client.Engineering/issues/2434.

**Pull request merged**